gaj23/fitlit-starter-kit
# Tm/feature/iteration1 data user repo

### What does this PR do? 
>Clearly and concisely describe the feature. 
- Create folder for test & create files for UserRepository class and test
- Add requires
- Add describe block
- Merge local main branch into another local branch to be updated with main
- Create test & pass for instances
- Add test-data. js file to use when testing for TDD
- Add data to test data file and required it in the test file
- Add getUser function and add test
- Create test for specific user and avgStep goals and added the methods to the class
- Add sad path testing to not find a specific user

### Is this a feature or refactor?  
Feature

### Problems Encountered & Resulting Solutions  
>State any problems encountered 
>Describe your code changes in detail for reviewers.

none for the code but, personally took a while to work through `calculateAvgStepGoal()`.

### Where should the reviewer start?
>What file(s) and line(s) are where changes are made?

**In src directory**
UserRepository.js: lines 1 - 24,

**In the test directory**
UserRepository-test.js: lines 1 - 34,

**In the data directory**
test-data.js: lines 1 - 74,

### How should this be tested?
>What steps need to be taken to confirm functionality? 

git fetch the branch, then in terminal run npm test and all the test should be passing

### Steps To Take
>Any questions and possible steps moving forward.

Think about if we might need to add any more sad path test or other test.
